homeassistant/components/ted/sensor.py

```diff
@@ -0,0 +1,196 @@
+"""Support gathering ted5000 and ted6000 information."""
+import logging
+
+from tedpy import MtuType, SystemType
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    Platform,
+    UnitOfElectricPotential,
+    UnitOfEnergy,
+    UnitOfPower,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import COORDINATOR, DOMAIN, NAME, OPTION_DEFAULTS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def build_sensor_descs(name: str, prefix: str, stype: str, is_net: bool):
+    """Return a list of sensors with given key prefix and type (Production / Consumption)."""
+    if is_net:  # If the sensor represents a net energy
+        total_state_class = SensorStateClass.TOTAL
+    else:
+        total_state_class = SensorStateClass.TOTAL_INCREASING
```

Comment:
I don't see how this works.

If the sensor is a net meter, and it's a daily or month to data type sensor, calculating a total requires `last_reset` to be set, but I don't see that in the code.
Please read this carefully: https://developers.home-assistant.io/docs/core/entity/sensor#entities-representing-a-total-amount

Reply:
Thank you for taking a look at this @emontnemery. You are right... I was confused as to how the total state class worked. I took a read through the docs and hopefully the latest commit resolve this.